akka-actor/src/main/scala/akka/AkkaVersion.scala

```diff
@@ -29,24 +29,26 @@ object AkkaVersion {
    */
   @InternalApi
   private[akka] def require(libraryName: String, requiredVersion: String, currentVersion: String): Unit = {
-    val VersionPattern = """(\d+)\.(\d+)\.(\d+)(-(?:M|RC)\d+)?""".r
-    currentVersion match {
-      case VersionPattern(currentMajorStr, currentMinorStr, currentPatchStr, mOrRc) =>
-        requiredVersion match {
-          case requiredVersion @ VersionPattern(requiredMajorStr, requiredMinorStr, requiredPatchStr, _) =>
-            // a M or RC is basically in-between versions, so offset
-            val currentPatch =
-              if (mOrRc ne null) currentPatchStr.toInt - 1
-              else currentPatchStr.toInt
-            if (requiredMajorStr.toInt != currentMajorStr.toInt ||
-                requiredMinorStr.toInt > currentMinorStr.toInt ||
-                (requiredMinorStr == currentMinorStr && requiredPatchStr.toInt > currentPatch))
-              throw new UnsupportedAkkaVersion(
-                s"Current version of Akka is [$currentVersion], but $libraryName requires version [$requiredVersion]")
-          case _ => throw new IllegalArgumentException(s"Required version string is invalid: [$requiredVersion]")
-        }
-
-      case _ => // SNAPSHOT or unknown - you're on your own
+    if (requiredVersion != currentVersion) {
```

Comment:
So this is the change, adding the check guard

Reply:
Yeah, it's a cheap fix rather than complete fix for the problem as other projects should probably not require a specific RC.

Reply:
could you give some more context about this problem? how critical is this? does it mean that 10.1.10 can't be used with RC1?

Reply:
If you build Akka HTTP with the system property `akka.version` set to "2.6.0-RC1" this leads to a check `AkkaVersions.require("Akka HTTP", required = "2.6.0-RC1", actual = "2.6.0-RC1")` which doesn't work right now (and is what this fixes). Requiring a regular x.y.z version and running with an RC works fine already today.

In reality I don't expect there will be any release of a library requiring "2.6.0-RC1" though, so this fix only makes it possible to easily do one-off builds like that.